openpype/hosts/traypublisher/batch_parsing.py

```diff
@@ -0,0 +1,100 @@
+"""Functions to parse asset names, versions from file names"""
+import os
+import re
+
+from openpype.lib import Logger
+from openpype.client import get_assets, get_asset_by_name
+
+from openpype.pipeline import (
+    CreatorError
+)
+
+
+def get_asset_doc_from_file_name(source_filename, project_name,
+                                 version_regex, all_selected_asset_ids=None):
+    """Try to parse out asset name from file name provided.
+
+    Artists might provide various file name formats.
+    Currently handled:
+        - chair.mov
+        - chair_v001.mov
+        - my_chair_to_upload.mov
+    """
+    version = None
+    asset_name = os.path.splitext(source_filename)[0]
+    # Always first check if source filename is directly asset (eg. 'chair.mov')
+    matching_asset_doc = get_asset_by_name_case_not_sensitive(
+        project_name, asset_name, all_selected_asset_ids)
+
+    if matching_asset_doc is None:
+        # name contains also a version
+        matching_asset_doc, version = (
+            parse_with_version(project_name, asset_name, version_regex,
+                               all_selected_asset_ids))
+
+    if matching_asset_doc is None:
+        matching_asset_doc = parse_containing(project_name, asset_name,
+                                              all_selected_asset_ids)
+
+    if matching_asset_doc is None:
+        raise CreatorError(
+            "Cannot guess asset name from {}".format(source_filename))
+
+    return matching_asset_doc, version
+
+
+def parse_with_version(project_name, asset_name, version_regex,
+                       all_selected_asset_ids=None, log=None):
+    """Try to parse asset name from a file name containing version too
+
+    Eg. 'chair_v001.mov' >> 'chair', 1
+    """
+    log = _handle_log(log)
```

Comment:
I don't think one additional line deserve new function.
```suggestion
    if not log:
        log = Logger.get_logger(__name__)
```